src/cli/term.rs

```diff
@@ -0,0 +1,120 @@
+use std::io::{self, Write};
+
+pub(crate) use termcolor::Color;
+use termcolor::{ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+use crate::currentprocess::filesource::{Writer, WriterLock};
+use crate::process;
+
+#[derive(Copy, Clone, Debug)]
+pub enum Attr {
+    Bold,
+    ForegroundColor(Color),
+}
+
+pub trait Terminal: Writer {
+    fn fg(&mut self, color: Color) -> io::Result<()>;
+    fn bg(&mut self, color: Color) -> io::Result<()>;
+    fn attr(&mut self, attr: Attr) -> io::Result<()>;
+    fn reset(&mut self) -> io::Result<()>;
+    fn carriage_return(&mut self) -> io::Result<()>;
+}
+
+/// Disable all terminal controls on non-tty's.
+pub(crate) struct StandardStreamTerminal {
```

Comment:
I think this should be pub(super), not pub(crate) - the terminal abstraction is really related to the edge of the code, and shouldn't leak to the core code base; I note that it has leaked to CurrentProcess, which pub(super) would have prevented.

Reply:
There isn't a good way not to expose it. Because we need to get the terminal implementation in the current process. So we at least need to expose it for `CurrentProcess`.